functions/Invoke-DbaDbShrink.ps1

```diff
@@ -0,0 +1,345 @@
+            }
+            catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            # changing statement timeout to $StatementTimeout
+            if ($StatementTimeout -eq 0) {
+                Write-Message -Level Verbose -Message "Changing statement timeout to infinity"
+            }
+            else {
+                Write-Message -Level Verbose -Message "Changing statement timeout to $StatementTimeout minutes"
+            }
+            $server.ConnectionContext.StatementTimeout = $StatementTimeoutSeconds
+
+            $dbs = $server.Databases | Where-Object { $_.IsSystemObject -eq $false -and $_.IsAccessible }
+
+            if ($Database) {
+                $dbs = $dbs | Where-Object Name -In $Database
+            }
+
+            if ($ExcludeDatabase) {
+                $dbs = $dbs | Where-Object Name -NotIn $ExcludeDatabase
+            }
+
+            foreach ($db in $dbs) {
+
+                Write-Message -Level Verbose -Message "Processing $db on $instance"
+
+                if ($db.IsDatabaseSnapshot) {
+                    Write-Message -Level Warning -Message "The database $db on server $instance is a snapshot and cannot be shrunk. Skipping database."
+                    continue
+                }
+
+                $files = @()
+                if ($FileType -in ('Log','All')) {
+                    $files += $db.LogFiles
+                }
+                if ($FileType -in ('Data','All')) {
+                    $files += $db.FileGroups.Files
+                }
+
+                foreach($file in $files) {
+                    try {
+                        $startingSize = $file.Size / 1024
+                        $spaceUsed = $file.UsedSpace / 1024
+                        $spaceAvailableMB = ($file.Size - $file.UsedSpace) / 1024
+                        $desiredSpaceAvailable = [math]::ceiling((($PercentFreeSpace/100)) * $spaceUsed)
+                        $desiredFileSize = $spaceUsed + $desiredSpaceAvailable
+
+                        Write-Message -Level Verbose -Message "File: $($file.Name)"
+                        Write-Message -Level Verbose -Message "Starting Size (MB): $([int]$startingSize)"
+                        Write-Message -Level Verbose -Message "Space Used (MB): $([int]$spaceUsed)"
+                        Write-Message -Level Verbose -Message "Starting Freespace (MB): $([int]$spaceAvailableMB)"
+                        Write-Message -Level Verbose -Message "Desired Freespace (MB): $([int]$desiredSpaceAvailable)"
+                        Write-Message -Level Verbose -Message "Desired FileSize (MB): $([int]$desiredFileSize)"
+                    }
+                    catch {
+                        $success = $false
+                        Stop-Function -message "Shrink Failed: $($_.Exception.InnerException)" -EnableException $EnableException -ErrorRecord $_ -Continue
```

Comment:
This is wrong and should be updated...

Something like `Stop-Function -Message "Calculation failure" -Target $file -ErrorRecord $_`

1. All this seems to be doing is calculating values to be used in the shrink operation. 
2. If that actually does fail for some reason should not be allowed to continue the command. It could cause unintentional consequences.